src/components/Home/Highlights/HighlightsManager.tsx

```diff
@@ -1,142 +1,109 @@
+import { api } from "api"
 
 /**
  * Component that decides the content of the carousel.
  * In the future the logic bheind the highlights will be implemented here and it will pass
  * the correct data to the PolimiCarousel
  */
 export const HighlightsManager: FC = () => {
-    const [isLessonFound, setIsLessonFound] = useState(false)
-    const [data, setData] = useState<CarouselItem[]>([
-        {
-            id: 0,
-            type: WidgetType.ISEE,
-            date: "Mer 22/10/2022",
-            time: "8:00",
-            title: "Scadenza ISEE 2023",
-        },
-        {
-            id: 1,
-            type: WidgetType.ISEE,
-            date: "Mer 22/10/2022",
-            time: "8:00",
-            title: "Scadenza ISEE 2023",
-        },
-        {
-            id: 2,
-            type: WidgetType.ISEE,
-            date: "Mer 22/10/2022",
-            time: "8:00",
-            title: "Scadenza ISEE 2023",
-        },
-    ])
-
-    //function used to add a "0" to the day is this one is lower than 10 (ex. n=8 -> res=08)
-    const pad = (n: number) => {
-        return n < 10 ? "0" + n : n
-    }
+    const { loggedIn, userInfo } = useContext(LoginContext)
 
-    const days = ["Lun", "Mar", "Mer", "Gio", "Ven", "Sab", "Dom"]
+    const [widgets, setWidgets] = useState<CarouselItem[]>([])
 
     /**
-     * function to determine which is the next lesson according to the current date and time
-     * @param timetable is the API response
+     * This function gets as parameters the list of events extracted and then it filters it.
+     * @param events
      */
-    const findNextLecture = (timetable: Lecture[]) => {
-        /**
-         * sorting dates in ascending order, priority is the 'date_start' field.If there are two lecture with the same
-         * 'date_start' we check the 'date_end' field!
-         * @example
-         * lecture 1{date_start: "2022-12-04T08:15:00",date_end: "2022-12-04T13:15:00"}
-         * lecture 2{date_start: "2022-12-04T08:15:00",date_end: "2022-12-04T14:00:00"}
-         * So in this case lecture 1 has more priority than lecture 2
-         */
-        const sorted = timetable.sort((a, b) => {
-            let dateA = new Date(a.date_start).getTime()
-            let dateB = new Date(b.date_start).getTime()
-            if (dateA !== dateB) {
-                return dateA - dateB
-            } else {
-                dateA = new Date(a.date_end).getTime()
-                dateB = new Date(b.date_end).getTime()
-                return dateA - dateB
-            }
-        })
+    const extractNextEvents = (events: Event[]) => {
+        const filteredEvents = events.filter(x =>
+            checkEventType(x.event_type.typeId)
+        )
 
-        const now = new Date()
+        const tempEvents: CarouselItem[] = []
+        for (let i = 0; i < filteredEvents.length; i++) {
+            tempEvents.push(createWidget(filteredEvents[i]))
+        }
+        setWidgets(tempEvents)
+    }
```

Comment:
qui ci sono un po' di cose che cambierei, innanzitutto direi di adottare un approccio un po' più functional con meno side-effect, quando chiami una funzione che in qualche modo trasforma i dati passati in ingresso dovresti cercare il più possibile di restituire i tuoi dati e non avere side effects che modificano dati esterni al metodo (e.g chiamare uno state-setter come `setWidgets()`)
Quando i metodi devono avere side effect è utile chiamarli appositamente (tipo qualcosa come `setNextEvents` invece di `findNextEvents` che mi da più l'idea di restituire i prossimi eventi invece di metterli nello state)

Io riscriverei la cosa più o meno così
```ts
    /**
     * This function gets as parameters the list of events extracted and then it filters it.
     * @param events
     */
    const extractNextEvents = (events: Event[]) => {
        const filteredEvents = events.filter(x =>
            checkEventType(x.event_type.typeId)
        )

        return filteredEvents.map(e => createWidget(e))
    }
```
così è chiaro anche per typescript, questa funzione prende un `Event[]` e ti ridà i giusti `CarouselItem[]` e più avanti puoi fare qualcosa tipo
```ts
setWidgets(extractNextEvents(events))

// oppure per avere le cose meno innestate

const newWidgets = extractNextEvents(events)
setWidgets(newWidgets)
```

inoltre questa cosa può andare bene ma personalmente non vorrei più di una lezione nella sezione highlights, se no si perde l'immediatezza della cosa, quindi aggiungerei un filtro anche per assicurarmi che ci sia una sola lezione nell'elenco
andrei anche un passo avanti e direi di mostrare la lezione solo se è strettamente il prossimo evento, non ha troppo senso mostrarla più avanti se è tipo tra due giorni, mi immagino l'use-case del tipo, sei sul campus e ti chiedi "ok dove devo andare per la prossima lezione?" apri l'app e la vedi subito, altrimenti è un po' inutile, però fai come meglio credi tu